packages/data-stores/src/i18n/test/resolvers.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Internal dependencies
+ */
+import { getLocalizedLanguageNames } from '../resolvers';
+import { setLocalizedLanguageNames } from '../actions';
+
+describe( 'i18n resolvers', () => {
+	describe( 'getLocalizedLanguageNames', () => {
+		it( 'should set the localized language names', () => {
+			const iter = getLocalizedLanguageNames( 'en-us' );
+
+			expect( iter.next().value ).toEqual( {
+				type: 'API_FETCH',
```

Comment:
This appears to be testing against the internals of the `apiFetch` data control. Is there away to abstract it, like for example, to use `apiFetch()` to generate this action?

Reply:
I don't think so, unless we want to replicate the entire call to apiFetch (which isn't so much a test as a behavior lock-in). What do you think? Personally I find these tests weak to begin with.

Reply:
The thing is that in the code we're using `apiFetch()` but in the test, we're testing against `API_FETCH` action type. I'd assume that the code we're testing and the tests would match; while right now, the test is testing against an internal of the external `apiFetch()` function. 

In theory, if that implementation changes over time, our code will continue working, but the test will fail, and that's not what I'd expect to happen. So I tend to think on the contrary: these tests are actually more flexible and sustainable to future changes. But I'd appreciate it if you could clarify why you consider such tests weak, maybe I'm misunderstanding something.

Furthermore, isn't using the [apiFetch data control here](https://github.com/WordPress/gutenberg/blob/master/packages/data-controls/src/index.js#L27) just fine? It has the same signature as the action we're creating in this test?